docs/diagnostic-logs.md

```diff
@@ -35,38 +42,115 @@ Diagnostic logs are disabled by default. To enable diagnostic logs, follow these
+
+Therefore, SignalR service provides two kinds of collecting behaviors
+* **collect all**: collect logs in all connections 
+* **collect partially**: collect logs in some specific connections
+
+> To distinguish the connections between those collect logs and those don't collect logs, SignalR service will treat some client as diagnostic client based on the diagnostic client configurations of server and client, in which the diagnostic logs always get collected, while the others don't. For more details, see [collect partially section](#collect-partially). 
+
+#### Collect all
+
+Diagnostic logs are collected by all the connections. Take messaging logs for example. When this behavior is enabled, SignalR service will send a notification to server to start generating tracing ID for each message. The tracing ID will be carried in the message to the service, the service will also log the message with tracing ID.
+
+> Note that to ensure the performance of SignalR service, SignalR service doesn't await and parse the whole message sent from client, therefore, the client messages isn't get logged. But if the client is marked as a diagnostic client, then client message will get logged in SignalR service.
+
+##### Configuration guide
+
+To enable this behavior, check the checkbox in the *Types* section in the *Log Source Settings*.
+
+This behavior doesn't require you to update server side configurations. This configuration changing will always be sent to server automatically.
+
+#### Collect partially
+
+Diagnostic logs are **only** collected by [diagnostic clients](#diagnostic-client). All messages get logged including client messages and connectivity events in the diagnostic clients.
+
+> The limit of the diagnostic clients' number is 100. If the number of diagnostic clients exceeds 100, the outnumbered diagnostic clients will get throttled by SignalR service. The new but outnumbered clients will be failed to connect to SignalR service, and throw `System.Net.Http.HttpRequestException` which has message `Response status code does not indicate success: 429 (Too Many Requests)`, while the already connected ones work without getting impacted by the throttling policy.
+
+##### Diagnostic client
+
+Diagnostic client is a logical concept, any client can be a diagnostic client. The server controls which client can be a diagnostic client. Once a client is marked as a diagnostic client, all diagnostic logs will be enabled in this client. To set a client be a diagnostic client, see the [configuration guide](#configuration-guide-1) below.
+
+> Diagnostic clients works for both **collect all** and **collect partially** collecting behaviors. It has higher priority to collect logs.
+
+##### Configuration guide
+
+To enable this behavior, you need to configure service, server, client side.
+
+###### Service side <!-- omit in toc -->
+
+To enable this behavior, uncheck the checkbox for a specific log type in the *Types* section in the *Log Source Settings*. 
```

Comment:
This is not accurate according to the note above.

> Diagnostic clients works for both **collect all** and **collect partially** collecting behaviors. It has higher priority to collect logs.

Reply:
Maybe move the note to here?

Reply:
Why move to here? The note talks about diagnostic client, put it under Diagnostic Client section is ok?

Reply:
I mean these two paragraphs conflict with each other. Putting them together may be easier to understand.